### PR TITLE
base1: Build and validate the base1 tests

### PR DESCRIPTION
**Pull request by stefwalter**
Build the base1 tests similar to the webpack built HTML tests.
    
 * We have the tests in a javascript file.
 * The javascript is compiled and validated.
 * Fixed *lots* of issues in the test javascript now that it's validated.
 * Generate the HTML wrapper automatically

### CODE REVIEW

**Comment by dperpeet** [comment]
```
make: *** No rule to make target 'dist/base1/test-base64.js', needed by 'distdir'.  Stop.
make: *** Waiting for unfinished jobs....
```

**Comment by stefwalter** [comment]
I think that fixed the race building the dist/base1/test-*.js

File: src/base1/test-chan.js [hunk 1]
```diff
@@ -612,41 +577,41 @@ <h2 id="qunit-userAgent"></h2><ol id="qunit-tests"></ol>
...
+QUnit.asyncTest("transport options", function() {
+    assert.expect(3);
 
     var channel;
     $(mock_peer).on("recv", function(event, chan, payload) {
         if (chan) {
-            equal(typeof cockpit.transport.options, "object", "is an object");
-            deepEqual(cockpit.transport.options, {
+            assert.equal(typeof cockpit.transport.options, "object", "is an object");
+            QUnit.deepEqual(cockpit.transport.options, {
```
Comment (dperpeet) [inline src/base1/test-chan.js]: Is there a reason not to use assert here? This conflicts with the other uses.

Comment (dperpeet) [inline src/base1/test-chan.js]: Changing during merge

**Pull request closed without merging**